docs/observability/langwatch.md

```diff
@@ -0,0 +1,181 @@
+)
+
+# Run the agent (all interactions will be traced)
+async for event in runner.run_async(
+    user_id=user_id,
+    session_id=session_id,
+    new_message=types.Content(role="user", parts=[
+        types.Part(text="What is the weather in New York?")]
+    )
+):
+    if event.is_final_response():
+        print(event.content.parts[0].text.strip())
+```
+
+## Advanced Configuration
+
+### Using Decorators for Additional Context
+
+If you want to add additional context or metadata to your traces, you can optionally use the `@langwatch.trace()` decorator:
+
+```python
+import langwatch
+from google.adk import Agent, Runner
+from google.adk.sessions import InMemorySessionService
+from google.genai import types
+from openinference.instrumentation.google_adk import GoogleADKInstrumentor
+
+langwatch.setup(
+    instrumentors=[GoogleADKInstrumentor()]
+)
+
+@langwatch.trace(name="Google ADK Agent Run")
+def run_agent_interaction(user_message: str):
+    # Update the current trace with additional metadata
+    current_trace = langwatch.get_current_trace()
+    if current_trace:
+        current_trace.update(
+            metadata={
+                "user_id": "user_123",
+                "session_id": "session_abc",
+                "agent_name": "weather_agent",
+                "model": "gemini-2.0-flash-exp"
+            }
+        )
+
+    # Your agent execution code here
+    # ... agent execution code ...
+
+    return "Agent response"
+```
+
+## View Results in LangWatch
+
+Once your application is running with the instrumentation, you can view all traces, metrics, and evaluations in your LangWatch dashboard:
+
+- **Traces**: See detailed execution flows of your agents
+- **Metrics**: Monitor performance and usage patterns
+- **Evaluations**: Assess agent behavior and run experiments
+- **Alerts**: Set up monitoring and get notified of issues
+
+![LangWatch Dashboard](../assets/langwatch-dashboard.png)
+
+## Notes
+
+- You do **not** need to set any OpenTelemetry environment variables or configure exporters manually—`langwatch.setup()` handles everything.
+- The `@langwatch.trace()` decorator is **optional** - the OpenInference instrumentor will capture all ADK activity automatically.
+- For advanced configuration (custom attributes, endpoint, etc.), see the [LangWatch Python integration guide](https://docs.langwatch.ai/integration/python).
+
+## Troubleshooting
+
+- Make sure your `LANGWATCH_API_KEY` is set in the environment.
+- If you see no traces in LangWatch, check that the instrumentor is included in `langwatch.setup()` and that your agent code is being executed.
+- Ensure you have the correct Google API key set for Gemini access.
+- Check that the `openinference-instrumentation-google-adk` package is properly installed.
+
+## Support and Resources
+
+- [LangWatch Documentation](https://docs.langwatch.ai)
+- [LangWatch Google ADK Integration Guide](https://docs.langwatch.ai/integration/python/integrations/google-ai)
```

Comment:
Note (not related to this PR): Would it make more sense to move the [ADK Guide](https://docs.langwatch.ai/integration/python/integrations/google-ai) under `Frameworks` in the LangWatch docs rather than the `Model Providers` section?